### PR TITLE
[BEAM-443] Remove BlockingDataflowRunner

### PR DESCRIPTION
**Pull request by peihe**

### CODE REVIEW

**Comment by peihe** [comment]
R: @bjchambers

File: examples/java/src/main/java/org/apache/beam/examples/common/ExampleUtils.java [hunk 1]
```diff
@@ -281,21 +276,7 @@ private void deletePubsubSubscription(String subscription) throws IOException {
...
-        && runner.equals(BlockingDataflowRunner.class)) {
-      // In order to cancel the pipelines automatically,
-      // {@literal DataflowRunner} is forced to be used.
-      options.setRunner(DataflowRunner.class);
-    }
-  }
-
-  /**
-   * If {@literal DataflowRunner} or {@literal BlockingDataflowRunner} is used,
+   * If {@literal DataflowRunner} is used,
```
Comment (bjchambers) [inline examples/java/src/main/java/org/apache/beam/examples/common/ExampleUtils.java]: Why is this restricted to the DatfalowRunner? I thought all PipelineResult objects allowed "waitUntilFinish"?

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowJobExecutionException.java [hunk 2]
```diff
@@ -20,7 +20,7 @@
 import javax.annotation.Nullable;
 
 /**
- * Signals that a job run by a {@link BlockingDataflowRunner} fails during execution, and
+ * Signals that a job run by a {@link DataflowRunner} fails during execution, and
```
Comment (bjchambers) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowJobExecutionException.java]: "fails" -> "failed"

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineJob.java [hunk 3]
```diff
@@ -257,6 +276,19 @@ State waitUntilFinish(
...
+          if (state == State.UPDATED) {
+            LOG.info("Job {} has been updated and is running as the new job with id {}."
+                + "To access the updated job on the Dataflow monitoring console, "
+                + "please navigate to {}",
+                getJobId(),
+                getReplacedByJob().getJobId(),
+                MonitoringUtil.getJobMonitoringPageURL(
+                    getReplacedByJob().getProjectId(), getReplacedByJob().getJobId()));
+          } else if (state == State.CANCELLED) {
+            LOG.info(String.format("Job %s cancelled by user", getJobId()));
```
Comment (bjchambers) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineJob.java]: Don't use String.format`. Instead Use `LOG.info("Job {} cancelled by user", getJobId())`

Also -- why are we telling the user the job was cancelled by the user? Maybe "Job {} finished with status CANCELLED"?

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineJob.java [hunk 4]
```diff
@@ -257,6 +276,19 @@ State waitUntilFinish(
...
+          } else if (state != State.DONE) {
+            LOG.info(String.format("Job %s failed with status %s", getJobId(), state));
```
Comment (bjchambers) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineJob.java]: Don't use `String.format`

File: examples/java/src/main/java/org/apache/beam/examples/common/ExampleUtils.java (hunk 1 above)
Comment (peihe) [inline examples/java/src/main/java/org/apache/beam/examples/common/ExampleUtils.java]: That part is sent out in https://github.com/apache/incubator-beam/pull/771

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowJobExecutionException.java (hunk 2 above)
Comment (peihe) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowJobExecutionException.java]: done

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineJob.java (hunk 3 above)
Comment (peihe) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineJob.java]: done

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineJob.java (hunk 4 above)
Comment (peihe) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineJob.java]: done

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineRegistrar.java [hunk 5]
```diff
@@ -34,28 +33,26 @@
...
+   * Register the {@link DataflowRunner}.
    */
   @AutoService(PipelineRunnerRegistrar.class)
   public static class Runner implements PipelineRunnerRegistrar {
     @Override
     public Iterable<Class<? extends PipelineRunner<?>>> getPipelineRunners() {
       return ImmutableList.<Class<? extends PipelineRunner<?>>>of(
-          DataflowRunner.class,
-          BlockingDataflowRunner.class);
+          DataflowRunner.class);
```
Comment (bjchambers) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineRegistrar.java]: May be worth a follow-up to modify PipelineRunnerRegistrar to only return a single PipelineRunner. Definitely not necessary in this PR.

File: runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/BlockingDataflowRunnerTest.java [hunk 6]
```diff
@@ -1,300 +0,0 @@
...
-
-  /**
-   * Tests that the {@link BlockingDataflowRunner} returns normally when a job terminates in
-   * the {@link State#DONE DONE} state.
-   */
-  @Test
-  public void testJobDoneComplete() throws Exception {
-    createMockRunner(createMockJob("testJobDone-projectId", "testJobDone-jobId", State.DONE))
-        .run(TestPipeline.create());
-    expectedLogs.verifyInfo("Job finished with status DONE");
```
Comment (bjchambers) [inline runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/BlockingDataflowRunnerTest.java]: I feel like these tests should have moved somewhere. Eg., where do we test that the right messages regarding the end state are logged when running a pipeline and waiting for it to complete? Where do we test that the right exceptions are thrown? etc.

Comment (peihe) [inline runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/BlockingDataflowRunnerTest.java]: The exceptions thrown was in BlockingDataflowRunner.run(), and they are removed.

I added tests for logging.

**Comment by peihe** [comment]
PTAL

**Comment by bjchambers** [comment]
LGTM

**Comment by davorbonaci** [comment]
Can we move forward on this PR? Is anything blocking it?

**Comment by peihe** [comment]
synced to head, and maven clean verify passed
@davorbonaci

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowJobExecutionException.java [hunk 7]
```diff
@@ -20,7 +20,7 @@
 import javax.annotation.Nullable;
 
 /**
- * Signals that a job run by a {@link BlockingDataflowRunner} fails during execution, and
+ * Signals that a job run by a {@link DataflowRunner} failed during execution, and
  * provides access to the failed job.
  */
 public class DataflowJobExecutionException extends DataflowJobException {
```
Comment (davorbonaci) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowJobExecutionException.java]: Seems unused. Perhaps it should go away, or be used.

Comment (peihe) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowJobExecutionException.java]: done

**Comment by peihe** [comment]
PTAL

**Comment by davorbonaci** [comment]
(ready to merge, once tests pass)

**Pull request closed without merging**